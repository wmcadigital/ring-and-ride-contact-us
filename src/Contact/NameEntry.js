import { useFormState } from "react-final-form";

import { required, name, composeValidators } from "../common/validation";
import FormSection from "../common/FormSection";
import Question from "../common/Question";
import ProgressIndicator from "../common/ProgressIndicator";
import TextInput from "../common/TextInput";

const NameEntry = () => {
  const stateApi = useFormState();

  const errorFirstName = stateApi.submitFailed
    ? stateApi.errors?.firstName
    : null;

  const errorLastName = stateApi.submitFailed
    ? stateApi.errors?.lastName
    : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 2 of 2"
        sectionName="About you"
      />
      <Question text="What is your name?" />
      <TextInput
        fieldName="firstName"
        label="First Name"
        validation={composeValidators(required, name)}
        error={errorFirstName}
      />
      <TextInput
        fieldName="lastName"
        label="Last Name"
        validation={composeValidators(required, name)}
        error={errorLastName}
      />
    </FormSection>
  );
};

export default NameEntry;
