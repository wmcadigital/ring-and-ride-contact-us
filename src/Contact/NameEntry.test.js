import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import NameEntry from "./NameEntry";

jest.mock("react-final-form");

describe("NameEntry", () => {
  it("renders as expected", () => {
    useFormState.mockImplementationOnce(() => ({}));
    const renderer = createRenderer();

    renderer.render(<NameEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      submitFailed: true,
      errors: { firstName: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<NameEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
