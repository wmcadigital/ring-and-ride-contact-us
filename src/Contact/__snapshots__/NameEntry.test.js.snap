// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameEntry renders as expected 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error={null}
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`NameEntry renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error="Required"
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;
